### ISSUE TITLE
Follow standard of clean git graph when using git.

### ISSUE THREAD

**Issue by nisrulz** [fossasia/open-event-android#460]
Right now as I look through the git history, I see everyone is making fixes in some branch and then when done is merging it into directly `master`. This approach is not correct.

I would like to propose that have a `develop` branch to which all merge happens be it from a PR or internal branch you have. Once the app is ready to release thats when you merge `develop` into `master` and hence forward you continue doing so. 

Doing so keeps the master branch clean and only with merge points having stable releases. 

As of now, the git graph is confusing as well as we mismanaged. There are unfinished issues that are being merged into `master` branch.

**Comment by the-dagger** [comment]
@nisrulz 

We have the following branches
master
All development goes on in this branch. If you're making a contribution, you are supposed to make a pull request to master. PRs to master must pass a build check and a unit-test (app/src/test) check on Travis
stable
Every few days (after some contributions have come to master), we merge master to stable. PRs from master -> stable will have to pass a build check, a unit-test check, AND a instrumentation test (app/src/androidTest) on Travis.
shipping
This contains shipped code. After significant features/bugfixes are accumulated on stable, we make a version update, and make a release. All tagged commits on shipping branch will automatically generate a release on Github with a copy of fDroid-debug and GooglePlay-debug apks.

**Comment by PareshMayani** [comment]
Yes this is the standard practice that should be followed for maintaining stable and development version.

**Comment by PareshMayani** [comment]
I mean what Nishant has proposed. Master branch should contain the stable code and develop branch should be used for development.

**Comment by mananwason** [comment]
I agree @nisrulz @PareshMayani. I also thought that we should keep master updated with stable code and different branches for shipping and development.

**Comment by nisrulz** [comment]
Reference Link : http://nvie.com/posts/a-successful-git-branching-model/#the-main-branches

**Comment by the-dagger** [comment]
Ok, sounds good. @nisrulz @PareshMayani .
I'll create develop branch and set up travis for it.
Thanks :+1:

**Comment by mananwason** [comment]
@the-dagger You can achieve the same by just renaming the branches. For example, rename the master to develop and so on

**Comment by the-dagger** [comment]
Lets create develop from the current master @mananwason.
As the shipping and stable are quite behind master

**Comment by mananwason** [comment]
Yeah cool. Sounds good. I believe you should create new shipping and stable, since the earlier ones would have a lot of conflicts with master.

**Comment by the-dagger** [comment]
Yeah.
I think that we won't be needing stable as we have master for that.

**Comment by nisrulz** [comment]
@mananwason @the-dagger The whole idea is to eliminate those `shipping` and `stable` branches.

This is what i propose.

1. Branch from `master` now and form `develop`.
1. Next branch more from `develop` if you need, but only merge back into  it when you are sure your `feature` branch code is completely working.
1. Next when you are done with all fixes or have reached your milestone, just merge into `master`.
1. Your `shipping` is represented by tagging it as a `release` 

As per my reference link, the whole idea is to work with only two main branches-  `master` and `develop`.

**Comment by mananwason** [comment]
@the-dagger stable and master would be different. master would contain the development code and stable is what has been tested.

**Comment by championswimmer** [comment]
rename 
master -> develop
stable -> master
shipping -> release

But nevertheless, git-flow does not make it compulsory to name the current branch `develop` and the stable one `master` 
We can name them whatever we wish to

If you want to make sure git clone gets you stable, then we can just set default branch to stable.

These are just semantics, and we are already following git-flow. What is in a name :wink:

**Comment by mananwason** [comment]
I think this should work @nisrulz. LGTM.

**Comment by championswimmer** [comment]
@mananwason @the-dagger 
If it really is not that necessary. keep it like it is. The travis and codecov configs will need changing too. 
We are **already** following a 3-branch git-flow strategy

**Comment by nisrulz** [comment]
@championswimmer naming is there to maintain standards for everyone else to be on the same page. As much as easy it is for normal devs to understand the deviations like this codebase has , its pretty much hard for others to decide where to start working from. 

Anyways you just re-stated my words , so all good. At the end of the day ,this is but a suggestion, no one is bound to follow semantics.

**Comment by championswimmer** [comment]
@nisrulz there is a readme file

**Comment by nisrulz** [comment]
@championswimmer if readme file was enough , you would not be having codacy running to check for coding standards. You could as go ahead and put things up in the readme for everyone to look up and follow the standards this project follows.

**Comment by championswimmer** [comment]
@nisrulz 
AOSP current development is on master. Releases are tagged. 

There are no **standards** of git branch names

**Comment by championswimmer** [comment]
So what are doing here will be 

stable -> development
shipping -> master

Dropping current master, as that branch does not have coverage, and we anyway merge after unit testing and building code. 

@the-dagger @mananwason please checkout/checkin code to existing branches so that you do not loose any data or have any conflicts. Let me know when there are no pending merges, so I can do this

**Comment by the-dagger** [comment]
Ok, wait.
Let us close/merge the existing Pull Requests.

**Comment by championswimmer** [comment]
Okay, this is done now.
